src/components/BarChart.js

```diff
@@ -0,0 +1,219 @@
+
+  getInitialState () {
+    return {
+      hovering: false,
+      milliseconds: 0
+    };
+  },
+
+  componentDidMount () {
+    this.setInterval(this.tick, 10);
+  },
+
+  componentWillReceiveProps () {
+    this.setState({
+      milliseconds: 0
+    });
+  },
+
+  tick () {
+    this.setState({
+      milliseconds: this.state.milliseconds + 10
+    });
+  },
+
+  _handleMouseOver (label, value, x, y) {
+    const animateDuration = 500;
+
+    this.props.onHover(label, value, x, y);
+
+    this.setState({
+      hovering: true,
+      milliseconds: this.props.animateOnHover ? animateDuration : this.state.milliseconds
+    });
+  },
+
+  _handleMouseOut () {
+    this.setState({
+      hovering: false
+    });
+  },
+
+  _handleOnClick () {
+    this.props.onClick(this.props.value);
+  },
+
+  render () {
+    const animateHeight = d3.ease('back-out', 0.5);
+    const height = this.props.height * animateHeight(Math.min(1, this.state.milliseconds / 1000));
+    const y = this.props.height - height + this.props.y;
+    const style = {
+      fill: this.state.hovering && this.props.hoverColor ? this.props.hoverColor : this.props.color,
+      cursor: 'pointer'
+    };
+
+    return (
+      <rect
+        height={height}
+        onClick={this._handleOnClick}
+        onMouseOut={this._handleMouseOut}
+        onMouseOver={this._handleMouseOver.bind(null, this.props.label, this.props.value, this.props.x, y)}
+        style={style}
+        width={this.props.width}
+        x={this.props.x}
+        y={y}
+      />
+    );
+  }
+});
+
+
+const BarChart = React.createClass({
+  propTypes: {
+    animateOnHover: React.PropTypes.bool,
+    color: React.PropTypes.string,
+    data: React.PropTypes.array.isRequired,
+    height: React.PropTypes.number,
+    hoverColor: React.PropTypes.string,
+    onClick: React.PropTypes.func,
+    onHover: React.PropTypes.func,
+    width: React.PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      animateOnHover: false,
+      color: StyleConstants.Colors.ASH,
+      height: 300,
+      hoverColor: StyleConstants.Colors.PRIMARY,
+      onClick: () => {},
+      onHover: () => {},
+      width: 500
+    };
+  },
+
+  shouldComponentUpdate (nextProps) {
+    return !_isEqual(nextProps, this.props);
+  },
+
+  _renderLabels (barWidth) {
+    const styles = this.styles();
+    const labels = this.props.data.map(d => {
+      return d.label;
+    });
+    const gap = 0.03;
+    const spaceBetweenBars = this.props.width * gap;
+    const totalWidth = barWidth * this.props.data.length + spaceBetweenBars;
+
+    return labels.map((label, index) => {
+      return (
+        <span
+          key={index}
+          style={Object.assign({}, styles.label, { width: totalWidth / this.props.data.length })}
+        >
+          {label}
+        </span>
+      );
+    });
+  },
+
+  render () {
+    const styles = this.styles();
+    const data = this.props.data.map(d => {
+      return d.value;
+    });
```

Comment:
Yes, the formatting of the data is left up to the parent component to handle and pass down to the `BarChart` component. The reason for `map` is because `.domain()` function on line 160 takes an array of values to calculate the yScale. `this.props.data` is an array of objects and each object has `label` and `value` properties.